Source/AMRInterpolator/IntegrationMethod.hpp

```diff
@@ -0,0 +1,68 @@
+/* GRChombo
+ * Copyright 2012 The GRChombo collaboration.
+ * Please refer to LICENSE in GRChombo's root directory.
+ */
+
+#ifndef INTEGRATIONMETHOD_HPP_
+#define INTEGRATIONMETHOD_HPP_
+
+#include <utility>
+#include <vector>
+
+//! A class to store and return the weights associated to a Newton-Cotes formula
+//! for numerical integration/quadrature which can be closed (i.e. includes the
+//! endpoints) or open (does not include the end points). This is used by
+//! SurfaceExtraction for integration over extraction surfaces.
+class IntegrationMethod
+{
+  private:
+    std::vector<double> m_weights;
+    int m_num_weights;
+    bool m_is_closed;
+
+  public:
+    //! Constructor
+    IntegrationMethod(const std::vector<double> &a_weights,
+                      bool a_is_closed = true)
+        : m_weights(a_weights), m_num_weights(a_weights.size()),
+          m_is_closed(a_is_closed)
+    {
+    }
```

Comment:
Maybe add an assertion checking that `a_weights.size()>0`

Reply:
Sure